pkg/util/azureclient/mgmt/containerservice/managedcluster_addons.go

```diff
@@ -6,19 +6,22 @@ package containerservice
 import (
 	"context"
 
-	mgmtcontainerservice "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-10-01/containerservice"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	armcontainerservice "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v6"
 )
 
 // ManagedClustersAddons is a minimal interface for azure ManagedClustersAddons
 type ManagedClustersAddons interface {
-	ListClusterAdminCredentials(ctx context.Context, resourceGroupName string, resourceName string, serverFqdn string) (mgmtcontainerservice.CredentialResults, error)
-	List(ctx context.Context) (mgmtcontainerservice.ManagedClusterListResultPage, error)
+	ListClusterAdminCredentials(ctx context.Context, resourceGroupName string, resourceName string) (armcontainerservice.ManagedClustersClientListClusterAdminCredentialsResponse, error)
+	List(ctx context.Context) *runtime.Pager[armcontainerservice.ManagedClustersClientListResponse]
 }
 
-func (r *managedClustersClient) ListClusterAdminCredentials(ctx context.Context, resourceGroupName string, resourceName string, serverFqdn string) (mgmtcontainerservice.CredentialResults, error) {
-	return r.ManagedClustersClient.ListClusterAdminCredentials(ctx, resourceGroupName, resourceName, serverFqdn)
+func (r *managedClustersClient) ListClusterAdminCredentials(ctx context.Context, resourceGroupName string, resourceName string) (armcontainerservice.ManagedClustersClientListClusterAdminCredentialsResponse, error) {
+	// TODO Determine if options need to be specified below
```

Comment:
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice#ManagedClustersClientListClusterAdminCredentialsOptions

I can't find any documentation properly explaining what the server FQDN parameter is for, and `az aks get-credentials -h` didn't help either (really poor description). But if the code works as-is, it's probably safe to remove the TODO comment.